app/src/main/java/com/github/k409/fitflow/data/UserRepository.kt

```diff
@@ -167,7 +172,19 @@ class UserRepository @Inject constructor(
             if (currentUser != null) {
                 val uid = currentUser.uid
                 val documentSnapshot = db.collection(USERS_COLLECTION).document(uid).get().await()
-                documentSnapshot.toObject(User::class.java)
+
+                User(
+                    uid = documentSnapshot.getLong("age")?.toString() ?: "",
+                    name = documentSnapshot.getString("name") ?: "",
+                    photoUrl = documentSnapshot.getString("photoUrl") ?: "",
+                    email = documentSnapshot.getString("email") ?: "",
+                    points = documentSnapshot.getLong("points")?.toInt() ?: 0,
+                    xp = documentSnapshot.getLong("xp")?.toInt() ?: 0,
+                    age = documentSnapshot.getLong("age")?.toInt() ?: 0,
+                    height = documentSnapshot.getDouble("height") ?: 0.0,
+                    weight = documentSnapshot.getDouble("weight") ?: 0.0,
+                    gender = documentSnapshot.getString("gender") ?: "",
+                )
```

Comment:
After recent changes "age" field in User class no longer exists. In this constructor it should be replaced by "dateOfBirth".
![image](https://github.com/PVP-K409/fit-flow/assets/127019248/1d75c2f9-4291-4f19-a88f-fc66991a6644)